diff --git a/node_modules/hexo-theme-butterfly/layout/includes/third-party/math/mermaid.pug b/node_modules/hexo-theme-butterfly/layout/includes/third-party/math/mermaid.pug
index f1d26b0..91ad316 100644
--- a/node_modules/hexo-theme-butterfly/layout/includes/third-party/math/mermaid.pug
+++ b/node_modules/hexo-theme-butterfly/layout/includes/third-party/math/mermaid.pug
@@ -274,8 +274,8 @@ script.
         if (!config.theme) {
           config.theme = theme
         }
-        const mermaidThemeConfig = `%%{init: ${JSON.stringify(config)}}%%\n`
-        const mermaidID = `mermaid-${index}`
+        const mermaidThemeConfig = '%%{init: ' + JSON.stringify(config) + '}%%\n'
+        const mermaidID = 'mermaid-' + index
         const mermaidDefinition = mermaidThemeConfig + mermaidSrc.textContent
 
         const renderFn = mermaid.render(mermaidID, mermaidDefinition)
@@ -308,12 +308,29 @@ script.
       })
     }
 
+    const preToMermaid = () => {
+      const preMermaidEle = document.querySelectorAll('pre.mermaid')
+      if (preMermaidEle.length === 0) return
+
+      preMermaidEle.forEach(ele => {
+        const preEle = document.createElement('pre')
+        preEle.className = 'mermaid-src'
+        preEle.hidden = true
+        preEle.textContent = ele.textContent
+        const newEle = document.createElement('div')
+        newEle.className = 'mermaid-wrap'
+        newEle.appendChild(preEle)
+        ele.replaceWith(newEle)
+      })
+    }
+
     const loadMermaid = () => {
+      preToMermaid()
       if (!{theme.mermaid.code_write}) codeToMermaid()
-      const $mermaid = document.querySelectorAll('#article-container .mermaid-wrap')
-      if ($mermaid.length === 0) return
+      const mermaidNodes = document.querySelectorAll('#article-container .mermaid-wrap')
+      if (mermaidNodes.length === 0) return
 
-      const runMermaidFn = () => runMermaid($mermaid)
+      const runMermaidFn = () => runMermaid(mermaidNodes)
       btf.addGlobalFn('themeChange', runMermaidFn, 'mermaid')
       window.loadMermaid ? runMermaidFn() : btf.getScript('!{url_for(theme.asset.mermaid)}').then(runMermaidFn)
     }
